### PR TITLE
498: Production Deploy on Release Publish

### PR DESCRIPTION
**Pull request by halprin**
# Production Deploy on Release Publish

Created a GitHub Action workflow that deploys to our production environment when a release is published in our GitHub repository.

## Issue

#498.

### CODE REVIEW

File: .github/workflows/prod-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Production Deploy
+
+on:
+  release:
+    types:
+      - published
```
Comment (jcrichlake) [inline .github/workflows/prod-deploy.yml:6]: Should this be on - published or on - released?  Seems like there are some small wrinkles as defined here 
https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=released#release

Comment (halprin) [reply .github/workflows/prod-deploy.yml:6]: `released`
> A release was published, or a pre-release was changed to a release.

`publish`
> A release, pre-release, or draft of a release was published.

Hmmm, it's a little hard to piece together.  Do you think we're missing the part when a pre-release is promoted to a release?  But I'm also not sure what we'd use a pre-release for in the first place.

Comment (jcrichlake) [reply .github/workflows/prod-deploy.yml:6]: I agree it's not the clearest difference.  The thing that raised my brow was a draft of a release being published, would that mean we would deploy a draft release when it's created in the UI?

Comment (halprin) [reply .github/workflows/prod-deploy.yml:6]: Ah, I did look into that.  When a draft of a release happens, then only `created` happens.  When the draft goes out of draft, it is `published`.

**Review by jorg3lopez — APPROVED** [review]
Great work!

**Pull request merged**